x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/host_isolation_exceptions_list.test.tsx

```diff
@@ -0,0 +1,106 @@
+  let waitForAction: AppContextTestRender['middlewareSpy']['waitForAction'];
+  beforeEach(() => {
+    getHostIsolationExceptionItemsMock.mockReset();
+    const mockedContext = createAppRootMockRenderer();
+    ({ history } = mockedContext);
+    render = () => (renderResult = mockedContext.render(<HostIsolationExceptionsList />));
+    waitForAction = mockedContext.middlewareSpy.waitForAction;
+
+    act(() => {
+      history.push(HOST_ISOLATION_EXCEPTIONS_PATH);
+    });
+  });
+  describe('When on the host isolation list page', () => {
+    const dataReceived = () =>
+      act(async () => {
+        await waitForAction('hostIsolationExceptionsPageDataChanged', {
+          validate(action) {
+            return isLoadedResourceState(action.payload);
+          },
+        });
+      });
+    describe('And no data exists', () => {
+      beforeEach(async () => {
+        getHostIsolationExceptionItemsMock.mockReturnValue({
+          data: [],
+          page: 1,
+          per_page: 10,
+          total: 0,
+        });
+      });
+
+      it('should show the Empty message', async () => {
+        render();
+        await dataReceived();
+        expect(renderResult.getByTestId('hostIsolationExceptionsEmpty')).toBeTruthy();
+      });
+    });
+    describe('And data exists', () => {
+      beforeEach(async () => {
+        getHostIsolationExceptionItemsMock.mockImplementation(getFoundExceptionListItemSchemaMock);
+      });
+      it('should show loading indicator while retrieving data', async () => {
+        let releaseApiResponse: (value?: unknown) => void;
+
+        getHostIsolationExceptionItemsMock.mockReturnValue(
+          new Promise((resolve) => (releaseApiResponse = resolve))
+        );
+        render();
+
+        expect(renderResult.getByTestId('hostIsolationExceptionsContent-loader')).toBeTruthy();
+
+        const wasReceived = dataReceived();
+        releaseApiResponse!();
+        await wasReceived;
+        expect(renderResult.container.querySelector('.euiProgress')).toBeNull();
+      });
+
+      it('should show items on the list', async () => {
+        render();
+        await dataReceived();
+
+        expect(renderResult.getByTestId('hostIsolationExceptionsCard')).toBeTruthy();
+      });
+      it('should show API error if one is encountered', async () => {
+        getHostIsolationExceptionItemsMock.mockImplementation(() => {
+          throw new Error('Server is too far away');
+        });
+        const errorDispatched = act(async () => {
+          await waitForAction('hostIsolationExceptionsPageDataChanged', {
+            validate(action) {
+              return isFailedResourceState(action.payload);
+            },
+          });
+        });
+        render();
+        await errorDispatched;
+        expect(
+          renderResult.getByTestId('hostIsolationExceptionsContent-error').textContent
+        ).toEqual(' Server is too far away');
+      });
+    });
+  });
+});
```

Comment:
strong nit?: The spacing between `it`, `describe` within this file looks odd to me. You add some new lines between the `it` blocks but in other cases you don't. Basically the lack and useage of newlines is inconsistent. I would make it consistent to give the next programmer the idea that this is a good set of tests.

Reply:
Funny enough it is not the first time I get this in a review. I personally don't mind the spacing between tests, my editor font size and font family handles that well.

I wonder if we could have an eslint rule for that. it is the kind of thing something like eslint --fix or prettier could handle without much hassle.